utils/src/main/scala/za/co/absa/enceladus/utils/general/Section.scala

```diff
@@ -0,0 +1,288 @@
+
+  if (length < 0) {
+    throw new InvalidParameterException(s"'length; cannot be negative, $length given")
+  }
+
+  if ((start >= 0) && (start.toLong + length.toLong > Int.MaxValue)) {
+    throw new IndexOutOfBoundsException(s"start and length combination are grater then ${Int.MaxValue}")
+  }
+
+  override def compare(that: Section): Int = {
+    if (start == that.start) {
+      length.compare(that.length) //shorter is smaller
+    } else if ((start < 0) && (that.start >= 0)) {
+      1 // negative start is bigger then the one of positive+zero
+    } else if ((start >= 0) && (that.start < 0)) {
+      -1
+    } else if (start.abs < that.start.abs) {
+      -1
+    } else {
+      1
+    }
+  }
+
+  /**
+    * Converts the Section to actual indexes representing the section on the given string if used in the substring function
+    * for example, e.g. first item of the tuple inclusive, second exclusive
+    * The result respects the boundaries of the string, the indexes not to cause OutOfBound exception
+    * @param forString  the string which the section would be applied to
+    * @return           tuple representing the beginIndex and endIndex parameters for the substring function
+    */
+  def toSubstringParameters(forString: String): (Int, Int) = {
+    val (realStart, after) = if (start >= 0) {
+        (start min forString.length, start + length)
+      } else {
+        val startIndex = forString.length + start
+        if (startIndex >= 0) { //
+          (startIndex, startIndex + length)
+        } else { // the distance from end is longer than the string itself
+          (0, length + startIndex max 0)
+        }
+      }
+    (realStart, after min forString.length)
+  }
+
+  /**
+    * The substring represented by this Section within the provided string
+    * Complementary to `remove`
+    * @param fromString the string to apply the section to
+    * @return           substring defined by this section
+    */
+  def extract(fromString: String): String = {
+    val (realStart, after) = toSubstringParameters(fromString)
+    fromString.substring(realStart, after)
+  }
+
+  /**
+    * Creates a string that is the remainder if the substring represented by this section is removed frm the provided string
+    * Complementary to `extract`
+    * @param fromString the string to apply the section to
+    * @return concatenation of the string before and after the Section
+    */
+  def remove(fromString: String): String = {
+    val (realStart, after) = toSubstringParameters(fromString)
+    fromString.substring(0, realStart) + fromString.substring(after)
+  }
+
+  /**
+    * Inverse function for `remove`, inserts the `what` string into the `into` string as defined by the `section`
+    * @param into the string to inject into
+    * @param what the string to inject
+    * @return     the newly created string
+    */
+  def inject(into: String, what: String): String = {
+
+    def invalidParameterExceptionMessage: String = {
```

Comment:
```suggestion
    def throwParameterException(): String = {
```
Definitely needs parens as the exception it throws is a side effect and this is most certainly not a getter of an exception message